README.md

```diff
@@ -184,12 +196,71 @@ Note: For Windows, CMake or Zig can be used.
     cmake --build . --config Release
     ```
 
-1. Use Zig
+- Using `Zig`:
 
     ```bash
     zig build -Drelease-fast
     ```
 
+### BLAS Build
+
+Building the program with BLAS support may lead to some performance improvements in prompt processing using batch sizes higher than 32 (the default is 512). BLAS doesn't affect the normal generation performance. There are currently three different implementations of it:
+
+- Accelerate Framework:
+
+  This is only available on Mac PCs and it's enabled by default. You can just build using the normal instructions.
+
+- OpenBLAS:
+
+  This provides BLAS acceleration using only the CPU. Make sure to have OpenBLAS installed on your machine.
+  
+  - Using `make`:
+    - On Linux:
+      ```bash
+      make LLAMA_OPENBLAS=1
+      ```
+      Note: In order to build on Arch Linux with OpenBLAS support enabled you must edit the Makefile adding at the end of the line 105: `-lcblas`
+
+    - On Windows:
+
+      1. Download the latest fortran version of [w64devkit](https://github.com/skeeto/w64devkit/releases).
+      2. Download the latest version of [OpenBLAS for Windows](https://github.com/xianyi/OpenBLAS/releases).
+      3. Extract `w64devkit` on your pc.
+      4. From the OpenBLAS zip that you just downloaded copy `libopenblas.a`, located inside the `lib` folder, inside `w64devkit\x86_64-w64-mingw32\lib`.
+      5. From the same OpenBLAS zip copy the content of the `include` folder inside `w64devkit\x86_64-w64-mingw32\include`.
+      6. Run `w64devkit.exe`.
+      7. Use the `cd` command to reach the `llama.cpp` folder.
+      8. From here you can run:
+
+          ```bash
+          make LLAMA_OPENBLAS=1
+          ```
+
+  - Using `CMake` on Linux:
```

Comment:
cmake should work everywhere, I don't think that we need to specify "on Linux" here

Reply:
I've specified it because I'm unable to find a guide that includes OpenBLAS on Windows using CMake. If there's a way to do it I'd be happy to include it in the README.

Reply:
I gave it a try by installing OpenBLAS with vcpkg, but it doesn't work, the CMakeLists.txt would need to be updated to support building with OpenBLAS on Windows. So I think this is fine for now.

Reply:
After installing OpenBLAS with vcpkg, it shows this:
```
openblas provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(OpenBLAS CONFIG REQUIRED)
    target_link_libraries(main PRIVATE OpenBLAS::OpenBLAS)
```
So maybe that would work better.

Reply:
```cmake
if (LLAMA_OPENBLAS)
    
	find_package(OpenBLAS)
	
    if (OpenBLAS_FOUND)
        message(STATUS "OpenBLAS found")

        add_compile_definitions(GGML_USE_OPENBLAS)
        set(LLAMA_EXTRA_LIBS ${LLAMA_EXTRA_LIBS} ${OpenBLAS_LIBRARIES})

        include_directories(${OpenBLAS_INCLUDE_DIRS})
    else()
        message(WARNING "OpenBLAS not found")
    endif()
endif()
```
This works with the OpenBLAS release .zip, if you pass the path to that on `CMAKE_PREFIX_PATH` but the CMake files that it comes with need to be edited to fix the paths inside.

Reply:
In Linux then it doesn't work because cblas is a separate library :disappointed_relieved:

Reply:
This works on Windows with vcpkg and seems clean enough:
```cmake
    find_package(OpenBLAS)
    if (OpenBLAS_FOUND)
        message(STATUS "OpenBLAS found")
        add_compile_definitions(GGML_USE_OPENBLAS)
        set(LLAMA_EXTRA_LIBS ${LLAMA_EXTRA_LIBS} OpenBLAS::OpenBLAS)
    else()
        message(WARNING "OpenBLAS not found")
    endif()
```
But for whatever reason, the Ubuntu package `libopenblas-dev` doesn't include the cmake module, even though it has been on the OpenBLAS repository [for 5 years](https://github.com/xianyi/OpenBLAS/blob/develop/cmake/OpenBLASConfig.cmake.in).

I give up, the performance difference isn't worth it anyway, and eventually our mat mul routines should be faster than OpenBLAS.

Reply:
There is the hard way, we make a `cmake/` subdirectory and add that to `CMAKE_MODULE_PATH` and inside we keep a bunch of scripts like `FindOpenBLAS.cmake` that just try all kinds of places to find the libraries, headers etc.